packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -307,6 +325,9 @@ class ChangeNotifier implements Listenable {
       _debugDisposed = true;
       return true;
     }());
+    if (_creationDispatched && MemoryAllocations.instance.hasListeners) {
```

Comment:
Is the expected use of the dispatchObjectEvent that you always check whether it hasListeners before you call it? We should document that.

Reply:
Also: Why does dispatchObjectEvent not just do this internally?